libapicore/ApiServer.cpp

```diff
@@ -142,63 +143,101 @@ void ApiConnection::processRequest(Json::Value& requestObject)
 	std::string _method = requestObject.get("method", "").asString();
 	jRes["id"] = requestObject.get("id", 0).asInt();
 
+	// Check authentication
+	if (!m_is_authenticated) {
+		if (_method == "api_authorize") {
 
-	if (_method == "miner_getstat1")
-	{
-		jRes["result"] = getMinerStat1();
-	}
-	else if (_method == "miner_getstathr")
-	{
-		jRes["result"] = getMinerStatHR();
-	}
-	else if (_method == "miner_shuffle")
-	{
-
-		// Gives nonce scrambler a new range
-		cnote << "Miner Shuffle requested";
-		jRes["result"] = true;
-		m_farm.shuffle();
+			if (!requestObject.isMember("params") || requestObject["params"].empty() || !requestObject["params"].isObject()) {
+				jRes["error"]["code"] = -32602;
+				jRes["error"]["message"] = "Missing params object member";
+			}
+			else {
+				Json::Value jPrm = requestObject["params"];
+				if (!jPrm.isMember("password") || jPrm["password"].empty() || !jPrm["password"].isString()) {
```

Comment:
I know - Claymore is not the standard.
So (just as info): Claymore uses `psw` as API parameter for the password.